src/snake.rs

```diff
@@ -11,13 +11,14 @@ pub enum Direction {
     Right,
 }
 
+#[derive(Clone, Copy, PartialEq)]
```

Comment:
why are this derives needed?

Reply:
Few things:
* PartialEq is needed so that we can use `==` comparison with blocks
* Clone is needed to implement copy
* Copy is not necessarily needed since we could use `&Block` in the signature of `collides_with` , but my linter told me that the `Block` structure is so small,  that it is inefficient to pass it by reference (&Block) because the pointer is longer than the two `i32` fields. So I passed it by value and since that consumes the Block, I needed copy so that it is automatically copied and I could continue to use  it afterwards.